docs/specifications/request_response_payloads.md

```diff
@@ -0,0 +1,118 @@
+# Lambda Request / Response Payload Specifications
+
+This tool will be deployed as an AWS Lambda, and the expected primary way of invoking it will be via HTTP `POST` requests.  The following details the structure of the request payload sent to the Lambda and the response payload returned.
+
+## Request Payload
+
+Example JSON payload sent via a `POST` request:
+```json
+{
+    "input_files": [
+        {
+            "s3_uri": "s3://my-bucket/apt-testing/dogs.tiff",
+            "filepath": "dogs.tiff",            
+            "checksums":{
+              "sha256": "23bcd2d83d4c0f270640ec65cbeb61a1784856255c3c98dd25ec340453458348s",
+            }
+        },
+				{
+            "s3_uri": "s3://another-bucket/apt-testing/CATS.PDF",
+            "filepath": "cats.pdf"            
+        },
+        {
+            "s3_uri": "s3://my-bucket/apt-testing/metadata.csv",
+            "filepath": "metadata/metadata.csv",            
+            "checksums":{
+              "md5": "9f81f3c07476a0d97f6793673dd8e475"
+            }
+        }
+    ],
+    "checksums":[
+      "md5",
+      "sha256"
+    ],
+    "output_zip_s3_uri": "s3://my-bucket/apt-testing/test-one-medium.zip",
+    "compress_zip": true
+}
+```
+### Fields
+
+- `input_files`: **REQUIRED** array of objects that define what files should be downloaded and added to the final bagit zipfile
+  - `s3_uri`: **REQUIRED** location to download the file from
+  - `filepath`: **REQUIRED** desired final filepath in the bagit zip file, relative to the root of the bag    - 
+    - e.g. `hello.txt` will end up at `/data/hello.txt` in the final bag, where `my/custom/path/goodbye.txt` would end up date`/data/my/custom/path/goodbye.txt`
+    - This provides a way to rename (filename) and organize (path) the file that was downloaded and written to the bag.
+  - `checksums`: **OPTIONAL** object of `<checksum name>: <checksum value>`
+    - If included, an error will be raised if the checksums generated during creation of the bag do not match this checksum
+    - Only checked if the checksum algorithm in this object is also included in the request root `checksums` field as well, instructing Bagit to generate a checksum for those algorithms
+    - If the checksum algorithm does not match a supported algorithm as defined below an error will be raised
+- `checksums`: **OPTIONAL** list of checksums for the bagit library to create for each file when creating the bag
```

Comment:
Is there a way to differentiate these 2 `checksum` fields to minimize potential confusion? Perhaps something like `input_file_checksum` in the array and `checksums_to_generate` for the bag, there are probably better options than those